platform/android/MapLibreAndroid/src/main/java/org/maplibre/android/storage/FileDirsPathsTask.kt

```diff
@@ -0,0 +1,25 @@
+package org.maplibre.android.storage
+
+import android.content.Context
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+class FileDirsPathsTask(
+	private val onCompletion: (resourcesCachePath: String, internalCachePath: String) -> Unit
+) {
+	fun execute(context: Context) {
+		CoroutineScope(Dispatchers.IO).launch {
+			try {
+				val path1 = FileSource.getCachePath(context)
+				val path2 = context.cacheDir.absolutePath
+				withContext(Dispatchers.Main) {
+					onCompletion(path1, path2)
+				}
+			} finally {
+				FileSource.unlockPathLoaders()
```

Comment:
Previously, this line was executed on the main thread. Now, it is executed on an IO thread (`AsyncTask::onCancelled` is called on the UI thread). Is this fine?

Reply:
It shouldn't matter which thread this gets called on.